drivers/include/w5500.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2023 Stefan Schmidt
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_w5500 W5500 ethernet driver
+ * @ingroup     drivers_netdev
+ * @brief       Driver for W5500 ethernet devices
+ *
+ * This device driver only exposes the MACRAW mode of W5500 devices, so it does
+ * not offer any support for the on-chip IPv4, UDP, and TCP capabilities of
+ * these chips. In connection with RIOT we are only interested in the RAW
+ * Ethernet packets, which we can use through netdev with any software network
+ * stack provided by RIOT (e.g. GNRC). This enables W5500 devices to communicate
+ * via IPv6, enables unlimited connections, and more...
+ *
+ * @note        This driver can be used in polling or interrupt mode.
+ *              On some shields the interupt line is not enabled by default,
+ *              you have to close the corresponding solder bridge to make the
+ *              interrupt mode work...
+ *
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for the W5500 device driver
+ *
+ * @author      Stefan Schmidt <stemschmidt@gmail.com>
+ */
+
+#ifndef W5500_H
+#define W5500_H
+
+#include <stdint.h>
+
+#include "periph/spi.h"
+#include "periph/gpio.h"
+#include "net/netdev.h"
+#include "kernel_defines.h"
+#include "ztimer.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   W5500 error codes
+ */
+enum {
+    W5500_ERR_BUS = -1,
+};
+
+/**
+ * @brief   W5500 device descriptor
+ */
+typedef struct {
+    spi_t spi;                    /**< SPI bus used */
+    spi_clk_t clk;                /**< clock speed used on the selected SPI bus */
+    gpio_t cs;                    /**< pin connected to the chip select line */
+    gpio_t evt;                   /**< pin connected to the INT line */
+    uint32_t polling_interval_ms; /**< interval for polling, 0 if interrupt mode */
+} w5500_params_t;
+
+/**
+ * @brief   Device descriptor for W5500 devices
+ */
+typedef struct w5500 {
+    netdev_t netdev;        /**< extends the netdev structure */
+    w5500_params_t p;       /**< device configuration parameters */
+    uint16_t frame_size_to_be_send;
+    uint16_t frame_size_sent;
```

Comment:
My understanding of confirm_send is that this function can be called at any time:
1. called before a new transmission is started (frame_size_to_be_send = 0): return the transferred size of the previous frame (frame_size_sent)
2. called after the new transmission has been started and is still in progress (frame_size_sent = 0): return -EAGAIN
3. called after the new transmission terminated -> see 1.

Reply:
I think the intention here was that this would only be called *after* the transmission and that the beginning of a new transmission would invalidate it (returns `-EAGAIN`).

So there is no need to store the size of the *previous* transmission.

But @maribu should know best.

Reply:
As @benpicco said:

1. Call `send()`, it returns right away, the hardware transmits in background
2. Wait for transmission to complete
3. Call `confirm_send()` exactly once to get the number of transmitted bytes or the error code

Reply:
Note: The is a provision for trivial polling:

```C
send();
while ((result = confirm_send) == -EAGAIN) {
    /* polling for TX to complete */
}
/* transmission result is in result */
```

Reply:
Ok, I understand. I thought you might be able to poll the current number of transmitted bytes of the frame while the transmission is still ongoing (the hardware might transmit the frame in chunks, so you could get the number of bytes already transmitted)... 
I will change the implementation.